From 57a500b1fbb72f5298702bc1f8424c9d91ac9cf8 Mon Sep 17 00:00:00 2001
From: Tram Nguyen <tram.nguyen.xw@rvc.renesas.com>
Date: Wed, 17 Aug 2016 21:10:31 +0700
Subject: [PATCH 5/6] gst145-v4l2bufferpool: Change the buffer number to be
 queued into V4L2

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

Apparently the continuous frame capturing mode of VIN has to
have at least 3 buffers to be queued into the driver in order to
output frames and the GStreamer pipeline occupies up to 2 buffers
, which correspond to the front and back buffers in the videosink.
The total number to circulate the V4L2 buffers in the pipeline is 5,
so this patch sets that as the buffer pool parameter.

Also this change suppresses the buffer copying that occurs when
running out of the remaining buffers in the queue.

v4l2bufferpool: Set the minimum number of the buffers as the same
with the maximum number

This patch pre-allocates the buffers as many as the maximum number
when the buffer pool is activated because the maximum number which
has already been set is essential for the contiguous capturing mode
and it would be efficient that all of the buffers are allocated
at the initialization of this plugin.

Signed-off-by: Tram Nguyen <tram.nguyen.xw@rvc.renesas.com>
---
 sys/v4l2/gstv4l2bufferpool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index dc44cdb..145fd43 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -1654,7 +1654,7 @@ gst_v4l2_buffer_pool_new (GstV4l2Object * obj, GstCaps * caps)
   gst_object_ref (obj->element);
 
   config = gst_buffer_pool_get_config (GST_BUFFER_POOL_CAST (pool));
-  gst_buffer_pool_config_set_params (config, caps, obj->info.size, 0, 0);
+  gst_buffer_pool_config_set_params (config, caps, obj->info.size, 5, 5);
   /* This will simply set a default config, but will not configure the pool
    * because min and max are not valid */
   gst_buffer_pool_set_config (GST_BUFFER_POOL_CAST (pool), config);
-- 
1.9.1

